rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -91,6 +92,11 @@ Node::create_subscription(
   const SubscriptionOptionsWithAllocator<AllocatorT> & options,
   typename MessageMemoryStrategyT::SharedPtr msg_mem_strat)
 {
+  // Setup topic statistics collector if enabled.
+  if (rclcpp::detail::resolve_enable_topic_statistics(options, *node_base_)) {
+    // TODO(@dabonnie, @prajakta-gokhale): https://github.com/ros2/ros2/issues/901.
+  }
```

Comment:
This should be done in the Subscription class instead no? That's where the intra process setting is resolved:

https://github.com/ros2/rclcpp/blob/911291f8d32da8a9a70f69adc7bcf1e41477e017/rclcpp/include/rclcpp/subscription.hpp#L140

Reply:
Initially that was my thought also. Synced with @dabonnie (he's working integrating all the pieces together in https://github.com/ros2/rclcpp/pull/1072) and he noticed that we can’t create the topic statistics object in `subscription.hpp` because of dependency compilation issues. So an instance of `SubscriptionTopicStatistics` has to be made outside and passed to Subscription.

If anything changes in his draft integration PR, we will move this where required.

Reply:
Ok, well how about we leave this out for now? Seems speculative and if he is going to update or potentially move this around, then we don't need this here right?

Reply:
Sounds good to me. I'll remove it from this PR.